### PR TITLE
fix: add confirmation box in resource summary delete action

### PR DESCRIPTION
**Pull request by ashutosh16**
When user delete the top-level resource like rollout/deployment/ingress. The user doesn't get confirmation box. This mismatch in experience cause the confusion with developer and may resulted in outage. If the resource is a top-level resource, the user should see a confirmation box to avoid overlooking.

Screenshot 2024-03-11 at 3 41 18 PM

This UI opens in treeView and resourceView

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

File: ui/src/app/applications/components/utils.tsx [hunk 1]
```diff
@@ -342,6 +347,12 @@ export const deletePopup = async (ctx: ContextApis, resource: ResourceTreeNode,
                 {isManaged ? (
                     <div className='argo-form-row'>
                         <FormField label={`Please type '${resource.name}' to confirm the deletion of the resource`} formApi={api} field='resourceName' component={Text} />
+                        <div className='caution-message-container'>
+                            <i className='fa fa-exclamation-triangle' style={{color: ARGO_FAILED_COLOR}} />
+                            <span style={{paddingLeft: '5px'}}>
+                                This action will delete a top-level resource and could cause a production <b>OUTAGES</b>.
+                            </span>
+                        </div>
```
Comment (crenshaw-dev) [inline ui/src/app/applications/components/utils.tsx:355]: Let's move this change to a separate PR

File: ui/src/app/applications/components/utils.tsx [hunk 2]
```diff
@@ -325,7 +325,12 @@ export const deletePodAction = async (pod: appModels.Pod, appContext: AppContext
 };
 
 export const deletePopup = async (ctx: ContextApis, resource: ResourceTreeNode, application: appModels.Application, appChanged?: BehaviorSubject<appModels.Application>) => {
-    const isManaged = !!resource.status;
+    function isTopLevelResource(res: ResourceTreeNode, app: appModels.Application): boolean {
+        const uniqRes = `/${res.namespace}/${res.group}/${res.version}/${res.kind}/${res.name}`;
+        return app.status.resources.some(resStatus => `/${resStatus.namespace}/${resStatus.group}/${resStatus.version}/${resStatus.kind}/${resStatus.name}` === uniqRes);
```
Comment (crenshaw-dev) [inline ui/src/app/applications/components/utils.tsx:330]:
```suggestion
        const uniqRes = `/${res.namespace}/${res.group}/${res.kind}/${res.name}`;
        return app.status.resources.some(resStatus => `/${resStatus.namespace}/${resStatus.group}/${resStatus.kind}/${resStatus.name}` === uniqRes);
```

For the purposes of determining "do we have a resource match," I think we can ignore version.

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

**Comment by ashutosh16** [comment]
I'll open new PR from my fork branch
